### PR TITLE
Fix build stuff

### PR DESCRIPTION
**Pull request by cevich**
(See commit comments)

### CODE REVIEW

File: subtests/docker_cli/build/build.py [hunk 1]
```diff
@@ -68,42 +69,22 @@ def reset_build_context(self):
...
-                    thing = img.long_id
-                else:
-                    thing = img.full_name
-                    # never ever remove base_repo_fqin under any circumstance
-                    if thing == base_repo_fqin:
-                        continue
-                self.logdebug("Removing image %s", img)
-                dcmd = DockerCmd(self, 'rmi', ['--force', thing])
-                dcmd.execute()
+            di = self.stuff['di']
```
Comment (edsantiago) [inline subtests/docker_cli/build/build.py]: May I assume you're aware of the KeyError here?

File: subtests/docker_cli/build/build.py [hunk 2]
```diff
@@ -483,93 +459,57 @@ def make_builds(self, source):
...
-                    if thing == base_repo_fqin:
-                        continue
-                self.logdebug("Removing image %s", img)
-                dcmd = DockerCmd(self, 'rmi', ['--force', thing])
-                dcmd.execute()
-
-
-class local_path(build_base):
+
+class local_path(BuildSubSubtest):
```
Comment (edsantiago) [inline subtests/docker_cli/build/build.py]: May I assume you're aware of the AttributeError? (I assume, but have not confirmed, that it's traceable to here)

**Comment by cevich** [comment]
> Looks like a good start. I won't be able to review again until Monday. Thanks again for taking this on!

Thanks, yep, np.

> May I assume

Assume nothing 😀

 I haven't run anything, was hoping the CI job would help out, but...seems it's broken...another thing to debug...

**Comment by cevich** [comment]
...there we go.  okay, there, it's failed properly now.  I'll pick this up again next week too.

**Comment by cevich** [comment]
jenkins, test this please.

**Comment by edsantiago** [comment]
I see a "test this please" but I don't see any commits fixing the KeyError and AttributeError bugs...?

**Comment by edsantiago** [comment]
I'm still seeing the bindir AttributeError

**Comment by cevich** [comment]
> I'm still seeing the bindir AttributeError

The CI job seems to agree with you...Perhaps I should actually fix the problem 😀

**Comment by cevich** [comment]
> I see a "test this please" but I don't see any commits fixing the KeyError and AttributeError bugs...?

Same here...some kind of odd github lag happening b/c I pushed to my branch.  Anyway, it's finally decided to show up ( 13f781a).

**Comment by cevich** [comment]
build	2 min 32 sec	Passed
build--garbage-check	1 sec	Regression

Lolz, okay, small progress I guess.  It seems there's a pair of orphan images that are leaking from somewhere.  To the debug logs Batman!

**Comment by cevich** [comment]
Okay, found it, ironically it's the "bad" sub-subtest that's leaking some orphan images.  There's also a bug where ``clean_all()`` is trying to remove those orphans by name (the empty string).

**Comment by cevich** [comment]
Found some missing commas in post-processing command lists.  Added a commit to fix that, account for the orphans during the ``bad`` sub-subtest, and clean them up after.  Not sure if it'll work or not, letting ADEPT CI check that out...

**Comment by edsantiago** [comment]
I just had a chance to spin up a virt and run tests. I'm seeing the same '...is not iterable' errors that ADEPT is reporting. Are you looking into that?

**Comment by cevich** [comment]
> Are you looking into that

Aye, It's probably some simple/dumb goof/typo I made somewhere.  I'll come back once I come up from air on the other trouble I've started...gaaaa...look at the time.  I'll tackle it next week, and snap...I probably should have a card for this too...

**Comment by cevich** [comment]
[(3) [CI] Fix build subtest's ability to count](https://trello.com/c/k5XEDA4O/508-3-ci-fix-build-subtest-s-ability-to-count)

**Comment by cevich** [comment]
Once again for the billionth time...(found newly undiscovered problem, added fix)...

**Pull request merged**